### DISCUSSION TITLE
Open an Editor in a dialog?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-platform/.github#73] [Q&A]
I'd like to open an editor in a dialog that normally is displayed as a part. In this case the `feature.xml` editor. Digging around in the code I just can't find an obvious way to do so, is there some way to accomplish this?

**Comment by merks** [comment]
Given org.eclipse.ui.IEditorPart.getEditorSite() -> org.eclipse.ui.IWorkbenchSite.getPage() I think an editor must be in a page so that it has access to the global menu bar and global toolbar (org.eclipse.ui.IEditorSite.getActionBars()) and such.  So no, I think there isn't even a non-obvious way to accomplish this.  You'll notice though that you can drag an editor to its own window...

**Reply by laeubi** [reply]
Can one have a separate blocking window?

https://wiki.eclipse.org/FAQ_How_do_I_open_an_editor_programmatically%3F

is just a bit leaving out the interesting parts :-)

**Comment by mickaelistria** [comment]
In a previous job, we managed to embed the groovy editor in a dialog, the code is available somewhere under https://github.com/bonitasoft/bonita-studio .
The interesting parts of an editor come from the SourceViewerConfiguraiton. So IIRC, rather than enabling a full editor (including various IDE-wide actions that do not make sense in a dialog), the trick was to create a source viewer and just use `.configure` with the right configuration. There were probably other tweaks involved, but that I think you can figure out.

**Comment by jonahgraham** [comment]
For embedding just a source viewer (as opposed to the whole editor), I think there are lots of example in JDT of that because that is how things like the formatter and code cleanup do it for their preference pages.

**Comment by laeubi** [comment]
A `feature.xml` Editor has a source tab, but provides much more complex edit capabilities. And because I do not officially "own" the implementation it is a bit hard to replicate all of its features by direct embedding individual editor capabilities.

**Comment by nitind** [comment]
Which features are you actually looking to offer? As an end-user of the feature editor, I don't understand what problems having it appear there would solve.

**Comment by Bananeweizen** [comment]
You could eventually reach the same goal by leaving the editor as is, but opening it on an EditorInput that's driven by some arbitrary data source you manage. As long as the editor itself doesn't do unreasonable casts to specific EditorInput (like FileEditorInput) that might be more simple. On save of the editor your data would be updated, and you could use that updated chunk wherever needed.

**Reply by laeubi** [reply]
The problem is that I'm in a dialog/wizard so even if it would work to open the part the user won't be able to edit the data in a useful manner.

**Comment by akurtakov** [comment]
In the early e4 days there was a css editor that was used in preferences dialog too but it required certain things in the eidtor to be done in the e4 way. I think https://marketplace.eclipse.org/content/eclipse-4-tools-lightweight-css-editor is the editor but where exactly it's sources are is smth I can't figure out after so many years.
